src/libs/ajax.test.js

```diff
@@ -79,4 +88,169 @@ describe('Ajax tests', () => {
+      method_version_id: '90000000-0000-0000-0000-000000000009',
+      wds_records: { record_type: 'FOO', record_ids: ['FOO1'] },
+      workflow_input_definitions: runSetInputDef,
+      workflow_output_definitions: runSetOutputDef
+    }
+
+    const body = JSON.stringify(payload)
+    const headers = { 'Content-Type': 'application/json' }
+
+    await cbasPact.addInteraction({
+      states: [
+        { description: 'ready to fetch recordId FOO1 from recordType FOO from wdsService' },
+        { description: 'ready to fetch myMethodVersion with UUID 90000000-0000-0000-0000-000000000009' },
+        { description: 'ready to receive exactly 1 call to POST run_sets' }
+      ],
+      uponReceiving: 'post a simple run set',
+      withRequest: { path: '/api/batch/v1/run_sets', method: 'POST', body, headers },
+      willRespondWith: { status: 200, body: expectedResponse }
+    })
+
+    await cbasPact.executeTest(async mockService => {
+      // ARRANGE
+      const signal = 'fakeSignal'
+
+      fetchCbas.mockImplementation(async path => await fetch(
+        `${mockService.url}/api/batch/v1/${path}`, { method: 'POST', body, headers }))
+
+      // ACT
+      const response = await Ajax(signal).Cbas.runSets.post(payload)
+
+      // ASSERT
+      expect(response).toBeDefined()
+      expect(fetchCbas).toBeCalledTimes(1)
+      expect(fetchCbas).toBeCalledWith('run_sets', { body, headers, method: 'POST', signal })
+      expect(response).toHaveProperty('run_set_id')
+      expect(response.runs.length).toEqual(1)
+    })
+  })
+
+  it('should successfully POST a run_set containing a "none" source', async () => {
+    const expectedResponse = {
+      run_set_id: fromProviderState('${run_set_id}', '00000000-0000-0000-0000-000000000000'), // eslint-disable-line no-template-curly-in-string
+      runs: [
+        {
+          run_id: fromProviderState('${run_id}', '00000000-0000-0000-0000-000000000000'), // eslint-disable-line no-template-curly-in-string
+          state: regex(RUN_STATE_REGEX, 'RUNNING'),
+          errors: regex('.*', 'some arbitrary string')
+        }
+      ],
+      state: regex(RUNSET_STATE_REGEX, 'RUNNING')
+    }
+
+    const payload = {
+      run_set_name: 'myRunSet',
+      run_set_description: 'myRunSet description',
+      method_version_id: '90000000-0000-0000-0000-000000000009',
+      wds_records: { record_type: 'FOO', record_ids: ['FOO1'] },
+      workflow_input_definitions: runSetInputDefWithSourceNone,
+      workflow_output_definitions: runSetOutputDef
+    }
+
+    const body = JSON.stringify(payload)
+    const headers = { 'Content-Type': 'application/json' }
+
+    await cbasPact.addInteraction({
+      states: [
+        { description: 'ready to fetch recordId FOO1 from recordType FOO from wdsService' },
+        { description: 'ready to fetch myMethodVersion with UUID 90000000-0000-0000-0000-000000000009' },
+        { description: 'ready to receive exactly 1 call to POST run_sets' }
+      ],
```

Comment:
Can you remind me again what is the purpose of these states and descriptions? It seems it is being repeated in all 3 tests that are added in this file

Reply:
Each interaction is allowed to specify any number of states that it needs for the interaction to work as intended. In this case, because we want to be able to call `POST run_sets`, we need there to be (1) records in the record table, (2) a methodVersion, and (3) to be able to verify the run set ID of the run set that will be returned in this interaction.

When CBAS runs its verification tests against these interactions, these states will tell it how to set itself up, using `@State` functions that match each of the above descriptions exactly.

The reason each of the tests has the same set of states is that they're all testing `POST run_sets` functionality - the same set of three states are needed to test each of the different `POST run_sets` calls that these tests run.

That we can reuse these states for many tests is a really good thing, because it means we don't need to have a 1:1 relationship between consumer tests and provider tests. The provider can just set up a bunch of states that many consumer tests can use multiple times.